broker/src/main/java/io/camunda/zeebe/broker/partitioning/Partition.java

```diff
@@ -134,24 +136,31 @@ public ActorFuture<Partition> leave() {
           raftPartition
               .leave()
               .whenComplete(
-                  (leaveOk, leaveError) -> {
-                    concurrencyControl.run(
-                        () -> {
-                          if (leaveError != null) {
-                            result.completeExceptionally(leaveError);
-                            return;
-                          }
-                          concurrencyControl.runOnCompletion(
-                              startupProcess.shutdown(concurrencyControl, context),
-                              (shutdownOk, shutdownError) -> {
-                                if (shutdownError != null) {
-                                  result.completeExceptionally(shutdownError);
-                                  return;
-                                }
-                                result.complete(this);
-                              });
-                        });
-                  });
+                  (leaveOk, leaveError) ->
+                      concurrencyControl.run(
+                          () -> {
+                            if (leaveError != null) {
+                              result.completeExceptionally(leaveError);
+                              return;
+                            }
+                            concurrencyControl.runOnCompletion(
+                                startupProcess.shutdown(concurrencyControl, context),
+                                (shutdownOk, shutdownError) -> {
+                                  if (shutdownError != null) {
+                                    result.completeExceptionally(shutdownError);
```

Comment:
:thought_balloon:
Is there a case where we leave, but fail to shut down the partition and don't want to delete the data? Just wondering. But I guess better be safe than sorry

Reply:
Not that I'm aware of but deleting data while the partition is still running could be bad so I think it's good that there's an early return when shutdown fails.